3152 docs rfc add aws private link to the virtual private cloud page (#3171)
timescale/docs: use-timescale/vpc.md

    *  `VPC`: select the VPC that is peered with Timescale.
    *  `Inbound rules`: leave empty.
    *  `Outbound rules`:
       * `Type`: `Custom TCP`
       * `Protocol`: `TCP`
       * `Port range`: `5432`
       * `Destination`: `Custom`
       * `Info`: the CIDR block of your Timescale VPC.
1.  Click `Add rule`, then click `Create security group`.

</Procedure>

### Attach a Timescale Service to the Peering VPC

Now that Timescale is communicating securely with your AWS infrastructure, you can attach 
one or more Timescale Services to the VPC. 

After you attach a Timescale Service to a VPC, you can only access it through the peered
AWS VPC. It is no longer accessible using the public internet.

<Procedure>

1.  In [Timescale Console > Services][console-services] select the Service you want to
    connect to the VPC.
1. Click `Operations` > `VPC`.
1. Select the VPC, then click `Attach VPC`.

</Procedure>

And that is it, your Timescale Service is now securely communicating with your AWS
account inside a VPC.

## Migrate a Timescale Service between VPCs

To ensure that your applications continue to run without interruption, you keep 

> Timescale Services attached to the VPC. However, you can change the VPN your 

Reviewer: ```suggestion
Timescale Services attached to the VPC. However, you can change the VPC your 
```

        - [vpc]
---

# Secure Timescale Services in a VPC

You use Virtual Private Cloud (VPC) peering to ensure that your Timescale Services are only accessible through
your secured AWS infrastructure. This reduces the potential attack vector surface and 
improves security.

<Highlight type="cloud" header="Sign up for Timescale" button="Try for free">
You are not charged for VPCs during your Timescale trial.
</Highlight>

> 

Reviewer 2: I would remove this. People assume the free trial is free unless someone tells them otherwise.

Author: Removed

        - [vpc]
---

# Securely connect to Timescale Services

You use Virtual Private Cloud (VPC) peering to ensure that your Timescale Services are 
only accessible through your secured AWS infrastructure. This reduces the potential 
attack vector surface and improves security.

<Highlight type="cloud" header="Sign up for Timescale" button="Try for free">
You are not charged for VPCs during your Timescale trial.
</Highlight>

The data isolation architecture that ensures a highly secure connection between your apps and Timescale:

<img class="main-content__illustration"
src="https://assets.timescale.com/docs/images/tsc-vpc-architecture.svg"
alt="The AWS Security Groups dashboard"/>

Your apps run inside your AWS Customer VPC, your Timescale Services always run 
inside the secure Timescale VPC. You control secure communication between apps in
your VPC and your Timescale Services using a dedicated Peering VPC. The AWS PrivateLink connecting 

> Timescale VPC to the dedicated Peering VPC gives a similar level of protection. 

Reviewer 2: ```suggestion
Timescale VPC to the dedicated Peering VPC gives the same level of protection as using a direct AWS PrivateLink connection. 
```

It only enables communication between your Customer VPC and your Timescale Services.

To configure this secure connection, you first create the Peering VPC with 
AWS PrivateLink in Timescale. After you have accepted and configured the 
peering connection to your Customer VPC, you use AWS Security Groups to 
restrict the services in your Customer VPC that are visible to the Peering VPC.
The last step is to attach individual Timescale Services to the Peering 

> Connection. 

Reviewer 2: ```suggestion
VPC. 
```

    src="https://assets.timescale.com/docs/images/tsc-vpc-create.png" 
    alt="Create a new Timescale VPC"/>

1.  For as many peering connections as you need:

    1. In the `VPC Peering` column, click `Add`.
    2. Enter information about your existing AWS VPC, then click `Add Connection`.

        <img class="main-content__illustration"
        src="https://assets.timescale.com/docs/images/tsc-vpc-add-peering.png"
        alt="Create a new Timescale VPC"/>

Timescale sends a peering request to your AWS account so you can 
[complete the VPC connection in AWS][aws-vpc-complete].
</Procedure>

### Complete the VPC connection in AWS
 
When you receive the Timescale peering request in AWS, edit your routing table to match 
the `IP Range` and `CIDR block` between your AWS and Timescale VPCs.

The request acceptance process is an important safety mechanism. Do not accept a
peering request from an unknown account.

<Procedure>

1. In [AWS > VPC Dashboard > Peering connections][aws-dashboard], select the peering connection 
    request from Timescale.

    Copy the peering connection ID to the clipboard. The connection request starts with `pcx-`.
 

> 1. In the peering connection, click  `Route Tables`, the select the `Route Table ID`

Reviewer 2: ```suggestion
1. In the peering connection, click  `Route Tables`, then select the `Route Table ID`
```

   <img class="main-content__illustration"
   src="https://assets.timescale.com/docs/images/aws-vpc-securitygroup.webp"
   alt="The AWS Security Groups dashboard"/>

    *  `VPC`: select the VPC that is peered with Timescale.
    *  `Inbound rules`: leave empty.
    *  `Outbound rules`:
       * `Type`: `Custom TCP`
       * `Protocol`: `TCP`
       * `Port range`: `5432`
       * `Destination`: `Custom`
       * `Info`: the CIDR block of your Timescale VPC.
1.  Click `Add rule`, then click `Create security group`.

</Procedure>

### Attach a Timescale Service to the Peering VPC

Now that Timescale is communicating securely with your AWS infrastructure, you can attach 
one or more Timescale Services to the VPC. 

After you attach a Timescale Service to a VPC, you can only access it through the peered
AWS VPC. It is no longer accessible using the public internet.

<Procedure>

1.  In [Timescale Console > Services][console-services] select the Service you want to
    connect to the VPC. 
   If you don't have a Timescale Service, [create a new one][create-service].
1. Click `Operations` > `VPC`.
1. Select the VPC, then click `Attach VPC`.

</Procedure>

> And that is it, your Timescale Service is now securely communicating securely with your AWS

Reviewer 2: ```suggestion
And that is it, your Timescale Service is now securely communicating with your AWS
```

account inside a VPC.

## Migrate a Timescale Service between VPCs

To ensure that your applications continue to run without interruption, you keep 
Timescale Services attached to the VPC. However, you can change the VPN your 
Timescale Service is attached to, or disconnect from a VPC and enable access to the 
Timescale Service from the public internet.

<Highlight type="info">
Timescale uses a different DNS for Timescale Services that are attached to a VPC.
When you migrate a Timescale Service between public access and a VPC, you need
to update your connection string.
</Highlight>

<Procedure>

1.  In [Timescale Console > Services][console-services] select the Service to migrate.

>     If you don't have a Timescale Service, [create a new one][create-service].

Reviewer 2: I don't think it makes sense to ask people to create a service if they don't have one in this section since these are the instructions to migrate an existing service

1. Click `Operations` > `VPC`.

> 1. Select the VPC, then click `Attach VPC`.

Reviewer 2: I'm not sure these instructions are 100% accurate since maybe you have to dettach the previous VPC?  Let's ask someone in engineering to confirm how this would work. Also to review the entire document. I don't know that all the details about how to set up AWS, etc. are still accurate or if things have changed since we created the documentation.
